### DISCUSSION TITLE
Is it possible to make field unique in a specific context? (unique by two fields)

### DISCUSSION THREAD

**Discussion by osbre** [filamentphp/filament#541] [Help]
Hello 👋.

In my migration file, I have these lines:

```php
$table->string('slug');
$table->foreignIdFor(School::class, 'school_id');

$table->unique(['slug', 'school_id']); // Unique by two fields.
```

And here is the field code:

```php
Components\TextInput::make('slug')
    ->unique('users', 'slug,school_id') // <- THIS
    ->required(),
```

How can I add "unique" check the same way as I did in my migration file?

**Comment by osbre** [comment]
Will try to use `addRules` method to add custom validation rule.
Update: doesn't seem that custom rule gets applied, potentially might be a bug

**Reply by danharrin** [reply]
Can you send the code you used for the custom rule? AFAIK Laravel doesnt support composite unique validation out of the box.
